src/config.rs

```diff
@@ -180,13 +180,22 @@ impl CacheConfigs {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize)]
+pub struct CustomToolchain {
+    pub compiler_executable: PathBuf,
+    pub archive: PathBuf,
+    pub archive_compiler_executable: String,
+}
+
```

Comment:
This is the way you specify a cross compiler toolchain - the compiler executable you want to override, an archive and a path within the archive you use to override the path of the compiler executable.

Reply:
OK, we can see how people like this in practice. I don't suppose it's worse than the current way people create cross-toolchains for icecream.